rust/src/measurements/discrete_laplace/linear/mod.rs

```diff
@@ -122,32 +131,43 @@ where
     D::Atom: Integer + SampleDiscreteLaplaceLinear<QO>,
     QO: Float + InfCast<D::Atom>,
 {
-    make_base_discrete_laplace_linear(scale, bounds)
+    make_base_discrete_laplace_linear(input_domain, input_metric, scale, bounds)
 }
 
 #[cfg(test)]
 mod tests {
-    use crate::domains::{AtomDomain, VectorDomain};
+    use crate::{
+        domains::{AtomDomain, VectorDomain},
+        metrics::AbsoluteDistance,
+    };
 
     use super::*;
 
     #[test]
-    fn test_make_discrete_laplace_mechanism_bounded() {
-        let measurement =
-            make_base_discrete_laplace_linear::<AtomDomain<_>, f64>(10.0, Some((200, 210)))
-                .unwrap_test();
+    fn test_make_discrete_laplace_mechanism_bounded() -> Fallible<()> {
+        let measurement = make_base_discrete_laplace_linear(
+            AtomDomain::<i32>::default(),
+            AbsoluteDistance::<i32>::default(),
+            10.0,
+            Some((200, 210)),
+        )?;
         let arg = 205;
-        let _ret = measurement.invoke(&arg).unwrap_test();
+        let _ret = measurement.invoke(&arg)?;
         println!("{:?}", _ret);
 
-        assert!(measurement.check(&1, &0.5).unwrap_test());
+        assert!(measurement.check(&1, &0.5)?);
+        Ok(())
     }
 
     #[test]
     fn test_make_vector_discrete_laplace_mechanism_bounded() {
-        let measurement =
-            make_base_discrete_laplace_linear::<VectorDomain<_>, f64>(10.0, Some((200, 210)))
-                .unwrap_test();
+        let measurement = make_base_discrete_laplace_linear(
+            VectorDomain::new(AtomDomain::default()),
+            Default::default(),
+            10.0,
+            Some((200, 210)),
+        )
+        .unwrap_test();
```

Comment:
You converted the other tests to `-> Fallible<()>` form, maybe this one too?

Also, could use a `measurement.check()`.